components/serializer.rst

```diff
@@ -628,6 +628,40 @@ processes::
     $anne = $normalizer->denormalize(['first_name' => 'Anne'], 'Person');
     // Person object with firstName: 'Anne'
 
+.. _using-underscored-method-names-for-camelized-attributes:
+
+snake_case to CamelCase
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Symfony provides a built-in name converter designed to transform between
+snake_case and CamelCased styles during serialization and deserialization
+processes::
+
+    use Symfony\Component\Serializer\NameConverter\SnakeCaseToCamelCaseNameConverter;
+    use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+
+    $normalizer = new ObjectNormalizer(null, new SnakeCaseToCamelCaseNameConverter());
+
+    class Person
+    {
+        public function __construct(
+            private string $fullName,
+        ) {
+        }
+
+        public function getFullName(): string
+        {
+            return $this->fullName;
+        }
+    }
+
+    $john = new Person('john_doe');
+    $normalizer->normalize($john);
+    // ['full_name' => 'johnDoe'];
+
+    $john = $normalizer->denormalize(['full_name' => 'johnDoe'], 'Person');
+    // Person object with fullName: 'jonh_doe'
```

Comment:
```suggestion
    class Person
    {
        public function __construct(
            private string $full_name,
        ) {
        }

        public function getFullName(): string
        {
            return $this->full_name;
        }
    }

    $john = new Person('john_doe');
    $normalizer->normalize($john);
    // ['fullName' => 'johnDoe'];

    $john = $normalizer->denormalize(['fullName' => 'johnDoe'], 'Person');
    // $john->getFullName() -> 'jonh_doe'
```

cc @xabbuh

Reply:
I am not sure if it works this way when the property is not public and the getter has a camel case name.

Reply:
Indeed, it's for camel_cased public properties and getters/setters (Eloquent-like).